### PR TITLE
Deal with duplicate lines

### LINKED ISSUE #8348: Suspicious repeated lines
Inspired by #8347, I looked for identical consecutive lines in CGAL headers. Some are completely normal, some are harmlessly redundant, some really look like they were supposed to be different.

https://github.com/CGAL/cgal/blob/847df9a0c9cbfee1168b4d3f969e082d431a39c9/Periodic_3_triangulation_3/include/CGAL/Periodic_3_regular_triangulation_3.h#L932-L934

https://github.com/CGAL/cgal/blob/847df9a0c9cbfee1168b4d3f969e082d431a39c9/Triangulation_2/include/CGAL/Triangulation_2/insert_constraints.h#L87-L88

etc. I think it would be worth it if someone could take a look to check that none are real bugs like the one in Triangulation.

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Deal with Issue #8348 

## Release Management

* Affected package(s):  Triangulation_2 and Periodic_triangulation_2
* Issue(s) solved (if any): fix #8348 
* License and copyright ownership:  unchanged

### CODE REVIEW

**Comment by afabri** [comment]
My latest commit left those open.


`````
=== "Algebraic_kernel_for_spheres/test/Algebraic_kernel_for_spheres/include/CGAL/_test_predicates.h" ===
109a110
>            theConstruct_2_3(20, 5, 0, 100),
=== "Arrangement_on_surface_2/benchmark/data/benchArrSegments.xml" ===
152a153
>       <benchArrCgalGmpqLazyCartesianSegment/>




=== "Arrangement_on_surface_2/include/CGAL/Arr_point_location/Td_active_fictitious_vertex.h" ===
134a136
>     if (v->is_at_open_boundary() && he->source()->is_at_open_boundary()) ++he;

=== "Arrangement_on_surface_2/include/CGAL/Arr_point_location/Td_traits.h" ===
1116a1130
>     if (v->is_at_open_boundary() && he->source()->is_at_open_boundary()) ++he;

=== "Arrangement_on_surface_2/include/CGAL/Curved_kernel_via_analysis_2/Arc_2.h" ===
569a574
>         _check_pt_arcno_and_coprimality(p, -1, c);

=== "Arrangement_on_surface_2/test/Arrangement_on_surface_2/data/circular_lines/compare" ===
153a154
> compare_xy    8 7 LARGER

=== "Arrangement_on_surface_2/test/Arrangement_on_surface_2/data/conics/compare" ===
153a154
> compare_xy    8 7 LARGER

=== "Arrangement_on_surface_2/test/Arrangement_on_surface_2/data/polylines/merge" ===
50a51
> are_mergeable 1 20 FALSE



=== "Boolean_set_operations_2/test/Boolean_set_operations_2/test_compilation.cpp" ===
233a236
>   gps.oriented_side(pgn_with_holes1);

AF:  (pgn_with_holes2 ??


=== "Circular_kernel_3/doc/Circular_kernel_3/CGAL/global_functions_spherical_kernel_3.h" ===
27a28
> \sa \link compare_xy_grp `CGAL::compare_xy()` \endlink
48a51
> \sa \link compare_xy_grp `CGAL::compare_xy()` \endlink
91a95
> \sa \link compare_xy_grp `CGAL::compare_xy()` \endlink


AF:  xyz ??





=== "Inscribed_areas/include/CGAL/Largest_empty_iso_rectangle_2.h" ===
1167a1184
>   --last_iter;

(comment: also is it guaranteed here thet the operator won't fail?)

=== "Installation/cmake/modules/CGAL_Macros.cmake" ===
18a19
>     set ( ${var} ${ARGN} CACHE INTERNAL "" )
22a24
>     set ( ${var} ${ARGN} CACHE ${type} ${doc} FORCE )

=== "Intersections_2/test/Intersections_2/test_intersections_2.cpp" ===
539a540
>     check_intersection     (S(p( 0,   0), p( 10,   0)), R(p(  1,   6), p(  1,  -3)), P(1, 0));
757a759
>     check_intersection<S>  (L(p( 18,  6), p( 0,   0)), Rec(p( 2,  0), p(6,  3)));



=== "Kernel_23/test/Kernel_23/include/CGAL/_test_cls_aff_transformation_3.h" ===
491a497
>  assert( FT( translate.hm(0,1) ) / FT( translate.hm(3,3) ) == FTzero );
503a510
>  assert( FT( translate.hm(3,0) ) / FT( translate.hm(3,3) ) == FTzero );





=== "Kernel_23/test/Kernel_23/include/CGAL/_test_new_2.h" ===
483a503
>   tmp34ab = CGAL::compare_distance(t2, l1, s1, p1);

````

File: BGL/test/BGL/test_Euler_operations.cpp [hunk 1]
```diff
@@ -17,7 +17,6 @@ test_copy_face_graph_nm_umbrella()
   T g;
   Kernel::Point_3 p(0,0,0);
 
-  CGAL::make_tetrahedron(p, p, p, p, g);
```
Comment (sloriot) [inline BGL/test/BGL/test_Euler_operations.cpp:20]: invalid change!

File: Distance_3/test/Distance_3/test_distance_3.cpp [hunk 2]
```diff
@@ -324,15 +324,15 @@ struct Test
...
+      check_squared_distance(S{p3, p2}, S{p5, p4}, 0);
 
       const double lambda_6 = r.get_double(0, 1);
       const P p6 = p3 + FT(lambda_6) * V{p3 - p2};
       // [p2;p3] overlaps [p5;p6]
       check_squared_distance(S{p2, p3}, S{p6, p5}, 0);
       check_squared_distance(S{p2, p3}, S{p5, p6}, 0);
       check_squared_distance(S{p3, p2}, S{p6, p5}, 0);
-      check_squared_distance(S{p3, p2}, S{p6, p5}, 0);
+      check_squared_distance(S{p3, p2}, S{p, p6}, 0);
```
Comment (sloriot) [inline Distance_3/test/Distance_3/test_distance_3.cpp:335]:
```suggestion
      check_squared_distance(S{p3, p2}, S{p5, p6}, 0);
```

File: BGL/test/BGL/test_Euler_operations.cpp (hunk 1 above)
Comment (afabri) [reply BGL/test/BGL/test_Euler_operations.cpp]: Why do we call it twice ?

Comment (sloriot) [reply BGL/test/BGL/test_Euler_operations.cpp]: To have 2 connected components

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-307

**Comment by afabri** [comment]
@efifogel can you please have a look at the duplicate lines in the arrangement package.

**Comment by afabri** [comment]
Td_active_fictitious_vertex.h   and Td_traits.h:    The lines are the same, but `he` gets incremented.

**Comment by afabri** [comment]
[Arc_2](https://github.com/CGAL/cgal/blob/2b3e0a709802ceb170d91b066161ad2d6fd4df7d/Arrangement_on_surface_2/include/CGAL/Curved_kernel_via_analysis_2/Arc_2.h#L573) needs in fact another look.  Should it  be `q`  instead of `p`  @efifogel  ?

**Comment by afabri** [comment]
In the Boolean operations [test ](https://github.com/CGAL/cgal/blob/2b3e0a709802ceb170d91b066161ad2d6fd4df7d/Boolean_set_operations_2/test/Boolean_set_operations_2/test_compilation.cpp#L237) I guess it should be `pgn_with_holes2`.  Do you agree @efifogel ?

**Comment by afabri** [comment]
[`CGAL_Macros.cmake`](https://github.com/CGAL/cgal/blob/2b3e0a709802ceb170d91b066161ad2d6fd4df7d/Installation/cmake/modules/CGAL_Macros.cmake#L18)  why the duplicate line here ?  @lrineau @sloriot

**Comment by sloriot** [comment]
> [`CGAL_Macros.cmake`](https://github.com/CGAL/cgal/blob/2b3e0a709802ceb170d91b066161ad2d6fd4df7d/Installation/cmake/modules/CGAL_Macros.cmake#L18) why the duplicate line here ? @lrineau @sloriot

no idea, even line 24 and 23 are identical

**Comment by lrineau** [comment]
> [`CGAL_Macros.cmake`](https://github.com/CGAL/cgal/blob/2b3e0a709802ceb170d91b066161ad2d6fd4df7d/Installation/cmake/modules/CGAL_Macros.cmake#L18) why the duplicate line here ? @lrineau @sloriot

That comes from commit 353864cf4abcd68a396d0ccda51d93b1618542bf, as a "workaround to SET bug in cmake 2.6". Anyway, `CGAL_Macros.cmake` is probably full of legacy CMake code.

**Comment by afabri** [comment]
Algebraic_kernel_for_spheres/test/Algebraic_kernel_for_spheres/include/CGAL/_test_predicates.h
A couple of lines further down we also have the same construction twice.   So I consider this normal.  Also an assert is done on the constructed object.

**Comment by afabri** [comment]
> > [`CGAL_Macros.cmake`](https://github.com/CGAL/cgal/blob/2b3e0a709802ceb170d91b066161ad2d6fd4df7d/Installation/cmake/modules/CGAL_Macros.cmake#L18) why the duplicate line here ? @lrineau @sloriot
> 
> That comes from commit [353864c](https://github.com/CGAL/cgal/commit/353864cf4abcd68a396d0ccda51d93b1618542bf), as a "workaround to SET bug in cmake 2.6". Anyway, `CGAL_Macros.cmake` is probably full of legacy CMake code.

But is this macro still used?   Then we would see if the problem is still there.

**Comment by afabri** [comment]
The Arrangement_on_surface_2/test/Arrangement_on_surface_2/data     are  fine.   It's just precomputed results to compare with.

**Pull request merged**